### ISSUE TITLE
Re-organize the mc file manager and editor pull-down menus

### ISSUE THREAD

**Issue by mc-butler** [MidnightCommander/mc#4205]
> [!IMPORTANT]
> This issue was migrated from Trac:
>
> | **Origin** | https://midnight-commander.org/ticket/4205 |
> | :------------- | :------------------------------------------------------------------------------------------------ |
> | **Reporter** | `ossi` (@ossilator) |

<p>
the editor's menus have grown over time into an incoherent mess and need a complete reorganization imo.<br />
</p>
<p>
this should be understood as a continuation of #320, #334, and #408.<br />
</p>
<p>
i'll come up with concrete proposals later unless someone beats me to it.<br />
</p>

**Comment by ossilator** [comment]
let's generalize this into "mc's menus" (as the linked issues mostly were).

**Comment by ossilator** [comment]
let's start with the main mc menu:

left/right menus:
- it's not really clear that the first section is a radio group. conventionally, one would add a checkmark on the left of the current mode's item, but that would require a whole new concept and feature set in the menus.
- the "encoding ..." item seems just wrong. is it actually side-specific? if so, then this is extremely confusing/error prone, and should be changed. the (consolidated) item belongs into the options menu.

file menu:
- it would be really a lot clearer to make the main f-keys (3-8) adjacent. these are a core aspect of the `nc` legacy, and should be treated as such.
- the "view file ..." entry is weird, and mostly useless. it should be available under shift-f3, kinda consistently with shift-f4. how to make it discoverable without it being in the menu is another question, but that's a pre-existing problem for other functions, and thus requires a solution regardless.
- the "quick cd" item doesn't fit, because it doesn't apply to the current entry. maybe it fits into left/right (because these have the navigation-related sections), otherwise command.
- the selection-related section doesn't really fit here. more conventionally, it would be in an "edit" menu. maybe "view". or "panel".
  - it's missing a plain "select current" item, which is kinda confusing.

command menu:
- the swap and switch panels items don't really belong in there. again something for the new "panel" (or whatever) menu.
- the "show directory sizes" item is kinda misplaced as well. firstly, it being a one-shot action, it should be named "calculate dir sizes" ("show" is conventionally used for toggles; see also the checkmark idea above). secondly, it doesn't pop up a window like the other actions, so it feels wrong. it might fit next to the "rescan" item in the left/right menus.
- intuitively, the "edit [...]" items all belong into the options menu

options menu:
- the "save setup" item kinda suggests that it refers only to the items above it, while in reality it also refers to the panel options. the help screen says so much, but it's still kinda confusing. ah, well.

then we have the new "about" item. like the "help" item (which is otherwise only in the keybar), it would kinda belong into an own "help" menu. otoh, this seems visually wasteful. i suppose it would make sense if there were more mildly related functions, e.g. quick search for actions (which in turn is kinda overkill for mc, but whatever). other than that, it would probably fit at the bottom of the command menu.

fwiw, i had a look at krusader's menu, and wasn't impressed, either.
maybe someone wants to have a look at `far` or total commander?

**Comment by zyv** [comment]
> * the "encoding ..." item seems just wrong. is it actually side-specific?

Yes, it is. But I don't see what's confusing about it. The idea was to enable panel encoding translation, for example, with old Windows FTP servers that return filename lists in CP1251, or with old FreeBSD FTP servers that return lists in KOI8-R, on a system that uses a different locale than the VFS locale (e.g., UTF-8). The relevance of all this today is another question, but 15 years ago, I used it all the time.

**Comment by ossilator** [comment]
ah. then "filename encoding" would make the scope much clearer.

i guess i would have made that an option in the connection dialogs and vfs urls.

otherwise it's not really clear how "sticky" the option is. alt-o/-i? ctrl-u? switching the location entirely?

**Comment by zyv** [comment]
> ah. then "filename encoding" would make the scope much clearer.
> 
> i guess i would have made that an option in the connection dialogs and vfs urls.

It is present in VFS URLs (but not explicitly as a selection in the connection dialogs)...

> otherwise it's not really clear how "sticky" the option is. alt-o/-i? ctrl-u? switching the location entirely?

... and as a consequence is "sticky" in the sense of alt-i / ctrl-u.

**Comment by egmontkob** [comment]
> the "view file ..." entry is weird, and mostly useless. it should be available under shift-f3, kinda consistently with shift-f4.

I fully agree it's a useless menu entry and a weird one there. I suggest to simply remove it. We don't have analogous "Edit file...", "Copy file..." etc. either that would operate on a typed filename rather than the one under the cursor.

Even without any other refactoring, this would make the File menu's Exit entry fit on 80x24, in presence of the Chattr entry (https://github.com/MidnightCommander/mc/issues/4903#issuecomment-3567755387 onwards). Not the right _solution_ for that issue per se, but coincidentally a fine workaround :)

(Mind you, Shift-F3 is different (view raw file), I wouldn't change its functionality.)

---

> it's not really clear that the first section is a radio group. conventionally, one would add a checkmark on the left of the current mode's item, but that would require a whole new concept and feature set in the menus.

`lib/widget/menu.c` already supports the currently unused `first_letter` for this purpose. Needs to be changed to `gunichar` and taken from the skin. Looks easily doable.

---

Another pet peeve of mine that's been bugging me for decades probably:

There's "Symlink", "Relative symlink", "Edit symlink". Couldn't they be combined to just 1, maybe 2 tops?

There's "Chmod" to chmod (also showing but not allowing to change the user/group), "Chown" to chown/chgrp (also showing but not allowing to change the perms), and there's "Advanced chown" for both, arbitrarily mentioning just one of the two features in its label. Again, ideally we should have at most 2 dialog boxes, maybe just 1. I don't think it's our job to provide two different user interfaces for performing the same task. If it really is, maybe a single dialog box could have an "Advanced mode" checkbox inside, switching its layout.
